### ISSUE TITLE
Provide clock parameter for COMB generators

### ISSUE THREAD

**Issue by dmurat** [f4b6a3/uuid-creator#60]
For easier testing with generated UUIDs, it would be great to have `getShortPrefixComb`, `getPrefixComb`, `getShortSuffixComb` and `getSuffixComb` variants that accept `java.time.Clock` parameter.
Tnx.

**Comment by fabiolimace** [comment]
I like it.

I will replace `System.currentTimeMillis()` with `SystemClock.millis()`. They have the same performance in benchmarks.

All JDK versions from 8 to 14 have the same implementation for the method `SystemClock.millis()` as follows:

```java
	@Override
	public long millis() {
		return System.currentTimeMillis();
	}
```

List of `SystemClock.millis()` implementations:

* [JDK8](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/time/Clock.java#l466)
* [JDK9](http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/d6a5cbeb8198/src/java.base/share/classes/java/time/Clock.java#l481)
* [JDK10](https://github.com/openjdk/jdk10/blob/6c31c0b4dc50495b89463730279512f851d9db0e/src/java.base/share/classes/java/time/Clock.java#L514)
* [JDK11](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/java/time/Clock.java#L514)
* [JDK12](https://github.com/openjdk/jdk12/blob/94d9355a640e7a51ec86fafbaf7dc8fb13570e1c/src/java.base/share/classes/java/time/Clock.java#L514)
* [JDK13](https://github.com/openjdk/jdk13/blob/dcd4014cd8a6f49a564cbb95387ad01a80a20bed/src/java.base/share/classes/java/time/Clock.java#L514)
* [JDK14](https://github.com/openjdk/jdk14/blob/abc56193174dd6a11a8453bd5036fc8618ecca61/src/java.base/share/classes/java/time/Clock.java#L515)

**Comment by fabiolimace** [comment]
Now the COMB factories have constructors with a `Clock` parameter for tests.

```java
public final class PrefixCombFactory extends AbstRandomBasedFactory {

	public PrefixCombFactory(Clock clock) {
		this(new DefaultRandomFunction(), clock);
	}

	public PrefixCombFactory(Random random, Clock clock) {
		this(getRandomFunction(random), clock);
	}

	public PrefixCombFactory(RandomFunction randomFunction, Clock clock) {
		super(UuidVersion.VERSION_RANDOM_BASED, randomFunction);
		this.clock = clock;
	}
```
The performance before and after the change is the same.

BEFORE:

```
Benchmark                     Mode  Cnt     Score    Error   Units
UuidCreator_PrefixComb       thrpt    5  2704,653 ± 53,146  ops/ms
UuidCreator_ShortPrefixComb  thrpt    5  2091,576 ± 51,236  ops/ms
```

AFTER:

```
Benchmark                     Mode  Cnt     Score    Error   Units
UuidCreator_PrefixComb       thrpt    5  2745,823 ± 53,917  ops/ms
UuidCreator_ShortPrefixComb  thrpt    5  2116,982 ± 27,708  ops/ms
```

Also added clock parameters to `DefaultTimeFunction` and `WindowsTimeFunction` for tests. The performance is the same too.

**Comment by fabiolimace** [comment]
Released version v4.4.1.

**Comment by dmurat** [comment]
Thank you, sir. You are very kind and fast :-)
